### DISCUSSION TITLE
Unsure if I can implement Docusaurus into my React Native Project instead of creating a seperate Directory For Docusaurus

### DISCUSSION THREAD

**Discussion by YusufGNR9** [facebook/docusaurus#9992] [General] [answered]
I am developing a React Native project and ı need to add documentation with docusaurus. I am able to understand docusaurus  documentation about starting a new project for documentation but The thing is We are a team and we need to keep track of changings on one place like we do for our React native project. My question is can I implement Docusaurus into my existing React Native project or should I create a seperate Directory for the documentation. Can anyone Enlighten me on this issue? I Really appreciate any assitance you can provide!

**Comment by slorber** [comment, marked as answer]
You should create a separate folder for the Docusaurus project. 

Your project/lib and Docusaurus do not have the same dependencies, so it's often not a good idea to try to fit everything into a single Node project. For example, your project might use React 17 while Docusaurus uses React 18, and this cause problems.

At the very least, have a dedicated folder + package.json for Docusaurus. I'd recommend using a monorepo setup (Yarn/npm workspace for example) but many users just create a subfolder and it also works.

You'll find many RN OSS libraries in our showcase with Docusaurus sites, to take a look at their setup:
- https://docusaurus.io/showcase
- https://github.com/Shopify/react-native-skia/tree/main/docs
- https://github.com/callstack/react-native-paper/tree/main/docs

**Comment by YusufGNR9** [comment]
I really Appreciate your help. I am gonna do this in this way creating seperate directory for docusaurus documentation. I was kinda lost in that process thank for enlighten me
